lua/lib/util.lua

```diff
@@ -228,4 +228,19 @@ function util.acronym(name)
   return (name:gsub("%s+", ""))
 end
 
+--- constrain a positive number to a positive min/max range
+-- @tparam number value
+-- @tparam number min
+-- @tparam number max
+-- @treturn number cycled value
+function util.cycle(value, min, max)
+  if value > max then
+    return util.cycle(value - max, min, max)
```

Comment:
the recursion trick is neat. but... it is pretty inefficient compared to a simple while-loop here. compare this version:

```
function util.cycle2(value, min, max)
    local y = value
    local d = max - min + 1
    while y > max do
      y = y - d
    end
    while value < min do
      y = y + d
    end
    return y
  end
  ```
  
  as the input gets really large, the second version runs much faster because it isn't adding function calls to the stack each time. (and eventually, the recursive version will eat all the stack; whereas with the non-recursive version we can add a counter for safety.)
  
  so this is one instance where i think using recursion isn't needed and isn't quite the right call.
  
  this is nitpicking, but its sort of important for a fundamental operation like this.

Reply:
oh wow this is way better